raft-tech/TANF-app
# Refine labels and text for Data Files search controls

## Summary of Changes
- Updated labels
- Updated error message
- Updated tests
Pull request closes #2699 

## How to Test
```
cd tdrs-frontend && docker-compose up --build
cd tdrs-backend && docker-compose up
```

1. Open http://localhost:3000/ and sign in.
2. Go to the Data Files page
3. Verify new paragraph text is there
4. Click the 'Search' button with various selections of the required fields and verify the error messages are correct along with the error count

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] Add the following paragraph text under the Data Files heading and before the first form input:
- [x] "Fields marked with an asterisk (*) are required."
- [x] Add required asterisks to all fields, including the STT field for OFA users, which should already have one.
- [x] Change the label for the Quarter dropdown to: "Fiscal Quarter*".
- [x] Ensure that existing form validation rules remain unchanged:
- [x] Only show relevant fields to each user group.
- [x] Only count visible errors in the error summary (fields not visible to the user should not be counted as errors).
- [x] Run the Testing Checklist to confirm all tests pass.
- [ ] Update the README if necessary, to reflect changes to the form or labels.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
